test/support.dart

```diff
@@ -18,13 +19,19 @@ Map<String, TreeBuilderFactory> get treeTypes {
   return _treeTypes;
 }
 
-final testDir = p.join(p.dirname(p.fromUri(Platform.packageConfig)), 'test');
-
-final testDataDir = p.join(testDir, 'data');
+Future<String> get testDirectory async {
+  final packageUriDir = p.dirname(p.fromUri(await Isolate.resolvePackageUri(
```

Comment:
nit: name this to `rootPackageDir` this sounds like the `lib` dir to me.

Reply:
This is the `lib` dir.

Below I have the comment about assuming the pub layout and an extra `dirname` to go from `lib` to the package root.

Reply:
Ohh right sorry just misreading